### PR TITLE
feature: multi-project installer

### PR DESCRIPTION
**Pull request by dustinbyrne**
If sub-projects are detected, this change will ask the user if they'd like to select sub-projects. If so, they'll have the ability to select which sub-projects to install to. The installer flow will then be run for each directory selected.

Also, incorrect installer path errors or process abort cases are now reported as `soft_failure`.

See the updated flow here: https://asciinema.org/a/e3sVXMwRHN8Ul1s4YKd10GMeg

### CODE REVIEW

File: packages/cli/src/cmds/agentInstaller/projectConfiguration.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+export interface ProjectConfiguration {
+  name: string;
+  path: string;
+  availableInstallers: AgentInstaller[];
+  selectedInstaller?: AgentInstaller;
+  writeAppMapYaml?: boolean;
+}
+
+export async function getProjects(
```
[the remark this answers is not in the document]
Comment (dustinbyrne) [reply packages/cli/src/cmds/agentInstaller/projectConfiguration.ts]: Thanks for the feedback. I typically don't break up linear code paths into separate functions unless something is called more than once. I know it's not a common approach, so I've gone ahead and broken it up a bit.

**Comment by dustinbyrne** [comment]
Thanks for pointing out the missing test. It looks like it got lost in the rebase, so I've brought it back.

File: packages/cli/src/cmds/agentInstaller/projectConfiguration.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dustinbyrne) [reply packages/cli/src/cmds/agentInstaller/projectConfiguration.ts]: Thanks for the feedback. I've gone ahead and broken it up a bit.

**Review by apotterri — APPROVED** [review]
Thanks for this. `getProjects` is easier to understand now, and the test is definitely a good thing.

FWIW, I think you could reasonably combine the two new commits with `feat: Add support for sub-project installation`.

**Pull request merged**